scripts/diff-pkg-changes.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/env python3
+
+# Authors: Dylan Turner
+# Description:
+# - Diff feeds of two NI Linux RT releases from their GitHub branches to create a json diff
+# - Steps:
+#     1. Pull core/ and extra/ feed info from GitHub for branches to compare
+#     2. Pull the Packages files from all four feeds
+#     3. Parse the package
+#     4. Compare the packages to each other
+#     5. Generate JSON report
+
+import sys
+from argparse import ArgumentParser, Namespace, RawTextHelpFormatter
+from requests import get, Response
+from json import dumps
+from dataclasses import dataclass
+
+# Defines the base URL for feeds
+DEF_FEED_BASE: str = 'http://download.ni.com/ni-linux-rt/feeds'
+
+# Constants used for constructing the url to a file on GitHub containing feed information
+GITHUB_BASE: str = 'https://raw.githubusercontent.com/ni/meta-nilrt'
+GITHUB_CONF_FNAME: str = 'conf/distro/nilrt.conf'
+ARCH: str = 'x64'
+SUB_FEEDS: list[str] = [ 'x64', 'core2-64', 'all' ]
+
+# If you change the info in the report, update this variable
+DATA_STRUCT_VERS = '0'
```

Comment:
Can you elaborate on this? I didn't follow it's usage.

Reply:
Because the data structure of the final report can change over time, we want to keep track of what "version" of the data structure we're currently using. If someone adds a field to the report, they should increase this value. It gets added to the report as `"version": DATA_STRUCT_VERS`, and since this is the initial version, so I put zero.